decoder/impeghd_obj_ren_dec.c

```diff
@@ -119,28 +119,20 @@ static IA_ERRORCODE impeghd_flex_spk_2_ls_geometry(ia_cicp_ls_geo_str *array_cic
   WORD32 i;
   WORD32 num_speakers;
   WORD32 ch_idx;
-  WORD32 cicp_idx = ptr_ref_spk_layout->cicp_spk_layout_idx;
   ia_flex_spk_data_str *ptr_flex_spk = &ptr_ref_spk_layout->str_flex_spk;
-  if (cicp_idx == 0 || cicp_idx == 8 || cicp_idx > NUM_LS_CFGS)
-  {
-    return IA_MPEGH_DEC_INIT_FATAL_INVALID_CICP_SPKR_INDEX;
-  }
-  num_speakers = impgehd_cicp_get_num_ls[cicp_idx];
-  *channel_names = ia_cicp_idx_ls_set_map_tbl[cicp_idx];
-  *num_channels = num_speakers;
-  *num_lfe = 0;
+  num_speakers = ptr_ref_spk_layout->num_speakers;
+
   if (ptr_ref_spk_layout->spk_layout_type == 2)
   {
     for (i = 0; i < num_speakers; i++)
     {
-      ch_idx = channel_names[0][i];
       pp_cicp_ls_geometry[i] = &array_cicp_ls_geometry[i];
-      pp_cicp_ls_geometry[i]->ls_elevation = ptr_flex_spk->str_flex_spk_descr[i].el_angle_idx;
-      pp_cicp_ls_geometry[i]->ls_elevation_start = ia_cicp_ls_geo_tbls[ch_idx].ls_elevation_start;
-      pp_cicp_ls_geometry[i]->ls_elevation_end = ia_cicp_ls_geo_tbls[ch_idx].ls_elevation_end;
-      pp_cicp_ls_geometry[i]->ls_azimuth = ptr_flex_spk->str_flex_spk_descr[i].az_angle_idx;
-      pp_cicp_ls_geometry[i]->ls_azimuth_start = ia_cicp_ls_geo_tbls[ch_idx].ls_azimuth_start;
-      pp_cicp_ls_geometry[i]->ls_azimuth_end = ia_cicp_ls_geo_tbls[ch_idx].ls_azimuth_end;
+      pp_cicp_ls_geometry[i]->ls_elevation = ptr_flex_spk->elevation[i];
```

Comment:
This doesn't look correct. elevation, elevation_start and elevation_stop are taking same value. Then what is the point of having three diff values?

Reply:
Since the speaker comes from flexible position, only the exact location is shared. hence, assigned all three fields with same value.